api/assets/allocate/index.php

```diff
@@ -17,12 +17,28 @@
 header('Content-Type: application/json');
 
 /// Builds the full address string from each part
+
 function GetFullAddress($line_1, $city, $region, $postcode) {
+
     // Check we have all info before proceeding
+
     if (empty($line_1) || empty($city) || empty($region) || empty($postcode)) {
+
         return NULL;
-    }
+       }
+    // Limits length of inputs, and returns NULL if excessive.
+    if ( (strlen($line)> 40) || (strlen($city)>35) || (strlen($region)>35) || (strlen($postcode)>9)){
+
+       return NULL;
```

Comment:
Since it's returning null, does any call to this function check for it being NULL?

Reply:
Good point, I'm not sure. It would be good to expect either a null or address returned from the function. I only did the `return NULL;` as that was already stated above, if any address line was empty.

Reply:
I'll fix it once I merge it in 😉